src/target/aby/trans.rs

```diff
@@ -0,0 +1,477 @@
+                }
+            }
+            Op::Const(Value::Bool(b)) => {
+                self.cache.insert(
+                    t.clone(),
+                    EmbeddedTerm::Bool(format!(
+                        "circ->PutCONSGate((uint64_t){}, (uint32_t){})",
+                        *b as isize, BOOLEAN_BITLEN
+                    )),
+                );
+            }
+            Op::Eq => {
+                let _s = self.embed_eq(t.clone(), &t.cs[0], &t.cs[1]);
+            }
+            Op::Ite => {
+                let sel = self.get_bool(&t.cs[0]).clone();
+                let a = self.get_bool(&t.cs[1]).clone();
+                let b = self.get_bool(&t.cs[2]).clone();
+                self.cache.insert(
+                    t.clone(),
+                    EmbeddedTerm::Bool(format!("circ->PutMUXGate({}, {}, {})", a, b, sel)),
+                );
+            }
+            Op::Not => {
+                let a = self.get_bool(&t.cs[0]);
+                self.cache.insert(
+                    t.clone(),
+                    EmbeddedTerm::Bool(format!("((BooleanCircuit *)circ)->PutINVGate({})", a)),
+                );
+            }
+            Op::BoolNaryOp(o) => {
+                let a = self.get_bool(&t.cs[0]).clone();
+                let b = self.get_bool(&t.cs[1]).clone();
+                match o {
+                    BoolNaryOp::Or => {
+                        self.cache.insert(
+                            t.clone(),
+                            EmbeddedTerm::Bool(format!(
+                                "((BooleanCircuit *)circ)->PutORGate({}, {})",
+                                a, b
+                            )),
+                        );
+                    }
+                    BoolNaryOp::And => {
+                        self.cache.insert(
+                            t.clone(),
+                            EmbeddedTerm::Bool(format!("circ->PutANDGate({}, {})", a, b)),
+                        );
+                    }
+                    BoolNaryOp::Xor => {
+                        self.cache.insert(
+                            t.clone(),
+                            EmbeddedTerm::Bool(format!("circ->PutXORGate({}, {})", a, b)),
+                        );
+                    }
+                }
+            }
+            Op::BvBinPred(op) => {
+                let a = self.get_bv(&t.cs[0]);
+                let b = self.get_bv(&t.cs[1]);
+
+                match op {
+                    BvBinPred::Uge => {
+                        let eq = self.embed_eq(t.clone(), &t.cs[0], &t.cs[1]);
+                        self.cache.insert(
+                            t.clone(),
+                            EmbeddedTerm::Bool(format!(
+                                "((BooleanCircuit *)circ)->PutORGate(circ->PutGTGate({}, {}), {})",
+                                a, b, eq
+                            )),
+                        );
```

Comment:
Did you consider embedding this (and ULT) as NOT composed with LT (and similar for ULT)? I worry that both EQ and LT/GT are expensive in MPC, while NOT is cheap.

Reply:
Will punt this to the next PR!